.github/scripts/pr_mod_file_tests.py

```diff
@@ -59,28 +59,45 @@ def _file_is_python(filename):
             is_python = True
         else:
             #If no ".py" extension exists, then
-            #open the file and look for a shabang
-            #that contains the word "python".
-            with open(filename, "r", encoding='utf-8') as mod_file:
-                #Loop over lines in file:
-                for line in mod_file:
-
-                    #Ignore blank lines:
-                    if line.strip():
-
-                        #Check that first non-blank
-                        #line is a shabang:
-                        if line[0:2] == '#!':
-                            #If so, then check that the word
-                            #"python" is also present:
-                            if line.find("python") != -1:
-                                #If the word exists, then assume
-                                #it is a python file:
-                                is_python = True
+            #try to open the file and look for
+            #a shabang that contains the word "python".
+            try:
+                with open(filename, "r", encoding='utf-8') as mod_file:
+                    #Loop over lines in file:
+                    for line in mod_file:
+
+                        #Ignore blank lines:
+                        if line.strip():
+
+                            #Check that first non-blank
+                            #line is a shabang:
+                            if line[0:2] == '#!':
+                                #If so, then check that the word
+                                #"python" is also present:
+                                if line.find("python") != -1:
```

Comment:
Would it be worthwhile to update this to the [recommended](https://peps.python.org/pep-0394/#for-python-runtime-distributors) pep8 specs to check for `python3`?

The following from the `pep8` doc might be the relevant reason if we are only supporting python3.7+:
> Some Linux distributions will not provide a `python` command at all by default, but will provide a `python3` command by default.

Reply:
I think in this case I would rather keep it as just `python`, because that way this script will catch all python files that are being committed (given that `python` as a string will match all the different variants).  This will ensure that even if by accident we commit a python2 file it will still be linted.

However, I agree that for our own python scripts we should make sure we are using `python3` instead of just `python` when using shebang calls.  I'll go ahead and modify this script's shebang as a start.

Reply:
@mwaxmonsky The three Github-related scripts should now be using `python3` instead of `python`.